### ISSUE TITLE
azure-functions:deploy fails if functionAppName has "." / project version #

### ISSUE THREAD

**Issue by tarinidash** [Microsoft/azure-maven-plugins#527]
### Plugin name and version

azure-maven-plugins

### Plugin configuration in your `pom.xml`
```
<version>1.0.0</version>

<properties>
<project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
        <maven.compiler.source>1.8</maven.compiler.source>
        <maven.compiler.target>1.8</maven.compiler.target>
        <azure.functions.maven.plugin.version>1.2.0</azure.functions.maven.plugin.version>azure-maven-plugins
        <azure.functions.java.library.version>1.2.0</azure.functions.java.library.version>
        <functionAppName>functions-${version}</functionAppName>
        <functionAppRegion>eastus</functionAppRegion>
        <stagingDirectory>${project.build.directory}/azure-functions/${functionAppName}</stagingDirectory>
        <functionResourceGroup>java-functions-group</functionResourceGroup>
</properties>

```
### Expected behavior
The deploy plugin should deploy the function to Azure 
### Actual behavior

[ERROR] Failed to execute goal com.microsoft.azure:azure-functions-maven-plugin:1.2.0:deploy (default-cli) on project functions: The host name functions-1.0.0.azurewebsites.net is invalid.: OnError while emitting onNext value: retrofit2.Response.class -> [Help 1]

### Steps to reproduce the problem

add project version to your functionAppName.

We would like to add version to the function so that we can see what version was deployed on azure. However if we add version the deployment fails.If we take out version(".") , the function deployment is successful.

Basically functionAppName cannot accept "."

**Comment by tarinidash** [comment]
Thanks @Flanker32 .

What is the best way to deploy a function with a version label?

**Comment by Flanker32** [comment]
@tarinidash  If you just want to add version to function name, can you just replace the `.` with `-`? Or you can add an attribute in the appSettings of maven plugin configuration in `pom.xml`(https://github.com/Microsoft/azure-maven-plugins/tree/develop/azure-functions-maven-plugin#configurations ),

**Comment by tarinidash** [comment]
@Flanker32 . Thanks for the feedback. We follow semiver(https://semver.org/) for all our deployments. I guess we have to skip it for azure functions.

The easiest way is to add pom version to the function name so that there is a single source of truth about the version.

exp: <functionAppName>functions-${version}</functionAppName>

If I add a new property or add a new appSettings, the developer has to make sure to change the version  in both places every time when there is a version upgrade/change. Its error prone too.

Let me know your thoughts about it.

**Comment by Flanker32** [comment]
@tarinidash 
You could also use ${project.version} in appSettings too, function plugin will help you set the attributes in development.
``` xml
<appSettings>
    ...
    <property>
        <name>APP_VERSION</name>
        <value>${project.version}</value>
    </property>
    ...
</appSettings>
```

**Comment by tarinidash** [comment]
Thanks @Flanker32 . I am closing this Issue.
